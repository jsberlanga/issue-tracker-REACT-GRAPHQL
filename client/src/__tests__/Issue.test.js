import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import IssueComponent, { SINGLE_ISSUE_QUERY } from "../components/Issue";
import { MockedProvider } from "react-apollo/test-utils";
import { BrowserRouter } from "react-router-dom";

const fakeIssue = {
  id: "123qwe",
  title: "Title for a Fake Issue",
  description: "Description for a Fake Issue",
  status: "OPEN"
};

describe("<Issue />", () => {
  it("renders and displays", () => {
    const wrapper = shallow(<IssueComponent issue={fakeIssue} />);
  });
});

describe("<Issue />", () => {
  it("renders with data", async () => {
    const mocks = [
      {
        request: { query: SINGLE_ISSUE_QUERY, variables: { id: "123qwe" } },
        result: {
          data: {
            issue: fakeIssue
          }
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <BrowserRouter>
          <IssueComponent issue={fakeIssue} />
        </BrowserRouter>
      </MockedProvider>
    );
    // console.log(wrapper.debug());
    expect(wrapper.text()).toContain("Fetching...");
    await wait();
    wrapper.update();
    // console.log(wrapper.debug());
    expect(toJSON(wrapper.find("h3"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("p"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("button"))).toMatchSnapshot();
  });
});
