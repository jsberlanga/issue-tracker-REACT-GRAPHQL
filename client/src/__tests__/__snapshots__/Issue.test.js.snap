// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Issue /> renders with data 1`] = `
<h3
  className="issue--title"
>
  Title: 
  Title for a Fake Issue
</h3>
`;

exports[`<Issue /> renders with data 2`] = `
Array [
  <p
    className="issue--description"
  >
    Description: 
    Description for a Fake Issue
  </p>,
  <p
    className="issue--status"
  >
    Current status: 
    <strong>
      OPEN
    </strong>
  </p>,
  <p
    className="issue--createdAt"
  >
    Created 
    NaN years ago
  </p>,
]
`;

exports[`<Issue /> renders with data 3`] = `
Array [
  <button
    className="btn__primary"
    onClick={[Function]}
  >
    DELETE
  </button>,
  <button
    className="btn__primary"
  >
    UPDATE
  </button>,
]
`;
